ap/tasks/plotting.py

```diff
@@ -99,252 +83,291 @@ def run(self):
+        b = self.branch_data
+        return self.local_target(f"plot__cat_{b.category}__var_{b.variable}.pdf")
 
     @PlotMixin.view_output_plots
     def run(self):
-        with self.publish_step("Hello from PlotShiftograms"):
-            import matplotlib.pyplot as plt
-            import mplhep
-            import hist
-            plt.style.use(mplhep.style.CMS)
+        import hist
+        import matplotlib.pyplot as plt
+        import mplhep
+
+        # prepare config objects
+        variable_inst = self.config_inst.get_variable(self.branch_data.variable)
+        category_inst = self.config_inst.get_category(self.branch_data.category)
+        leaf_category_insts = category_inst.get_leaf_categories() or [category_inst]
+        process_insts = list(map(self.config_inst.get_process, self.processes))
+        sub_process_insts = {
+            proc: [sub for sub, _, _ in proc.walk_processes(include_self=True)]
+            for proc in process_insts
+        }
+        shift_inst_up = self.config_inst.get_shift(f"{self.branch_data.shift_source}_up")
+        shift_inst_down = self.config_inst.get_shift(f"{self.branch_data.shift_source}_down")
+
+        # histogram data per process
+        hists = OrderedDict()
+
+        with self.publish_step(f"plotting {variable_inst.name} in {category_inst.name}"):
+            for dataset, inp in self.input().items():
+                dataset_inst = self.config_inst.get_dataset(dataset)
+                h_in = inp["collection"][0].load(formatter="pickle")[variable_inst.name]
+
+                # loop and extract one histogram per process
+                for process_inst in process_insts:
+                    # skip when the dataset is already known to not contain any sub process
+                    if not any(map(dataset_inst.has_process, sub_process_insts[process_inst])):
+                        continue
+
+                    # work on a copy
+                    h = h_in.copy()
+
+                    # axis selections
+                    h = h[{
+                        "process": [
+                            hist.loc(p.id)
+                            for p in sub_process_insts[process_inst]
+                            if p.id in h.axes["process"]
+                        ],
+                        "category": [
+                            hist.loc(c.id)
+                            for c in leaf_category_insts
+                            if c.id in h.axes["category"]
+                        ],
+                    }]
+
+                    # axis reductions
+                    h = h[{"process": sum, "category": sum}]
+
+                    # add the histogram
+                    if process_inst in hists:
+                        hists[process_inst] += h
+                    else:
+                        hists[process_inst] = h
+
+            # there should be hists to plot
+            if not hists:
+                raise Exception("no histograms found to plot")
```

Comment:
This code block is (almost) the same as in the class `PlotVariables`, right? At least at first glance 🤔  I wonder if the accessing and loading of the histograms can be somehow centralized, but I'm not sure myself :D

Reply:
Yup, except for the reduction of the shift axis they are quite similar. I think Mathis is working on this in a different PR :)